src/legacy/core_plugins/dashboard_embeddable_container/public/np_ready/public/lib/embeddable/dashboard_container_factory.ts

```diff
@@ -39,18 +48,9 @@ export class DashboardContainerFactory extends EmbeddableFactory<
   public readonly type = DASHBOARD_CONTAINER_TYPE;
   private allowEditing: boolean;
 
-  constructor({
-    savedObjectMetaData,
-    capabilities,
-  }: {
-    savedObjectMetaData?: SavedObjectMetaData<SavedObjectAttributes>;
-    capabilities: {
-      showWriteControls: boolean;
-      createNew: boolean;
-    };
-  }) {
-    super({ savedObjectMetaData });
-    this.allowEditing = capabilities.createNew && capabilities.showWriteControls;
+  constructor(options: DashboardOptions, private readonly viewportProps: ViewportProps) {
```

Comment:
nit: Not sure I like the name `viewportProps` because it's so coupled to the react component. We ever change the name of that and I think an engineer would be confused by what `viewport` means in this context.  Maybe something generic like `handlers` since they are all functions (Canvas team calls the an object of functions they pass `handlers`).

Reply:
Renamed to `containerOptions` as those are options for `DashboardContainer`.